### ISSUE TITLE
crop extent in `make_africa_mask` when `countries` is not `NULL`

### ISSUE THREAD

**Issue by geryan** [idem-lab/sdmtools#69]
e.g. the below is necessary becase the first call uses the extent for all Africa which is dumb

This function needs redoing wholesale I reckon. Come on mate lift your game.

```
make_africa_mask(
      type = "raster",
      res = "high",
      countries = "COD"
    ) |>
      crop(
        y = make_africa_mask(
          type = "vector",
          countries = "COD"
        )
      )
```

**Comment by geryan** [comment]
See #46

**Comment by geryan** [comment]
resolved by #72
